### PR TITLE
seleniumJsonCapabilities 

### PR DESCRIPTION
**Pull request by teunisnl**
#63 

Tried the included jsonhelper first, but that has issues with maps within maps
Not sure if you willing to use/include Gson, otherwise I have to write a parser.
Unfortunately you will have to escape all ", or add a replace ' by " inside the new Factory

Examples that I tested (launch the chrome correct)
```
mvn clean test-compile failsafe:integration-test -DseleniumGridUrl=http://localhost:4444/wd/hub -DseleniumJsonCapabilities="{\"browserName\":\"chrome\",\"chromeOptions\":{\"mobileEmulation\":{\"deviceName\":\"Google Nexus 5\"}}}" -DfitnesseSuiteToRun=HsacExamples.SlimTests.BrowserTests.ScriptTest

or

mvn clean test-compile failsafe:integration-test -DseleniumGridUrl=http://localhost:4444/wd/hub -DseleniumJsonCapabilities="{\"browserName\":\"chrome\",\"chromeOptions\":{\"mobileEmulation\":{\"deviceMetrics\":{\"pixelRatio\":3.0,\"width\":360,\"height\":640},\"userAgent\":\"Mozilla/5.0 (Linux; Android 4.2.1; en-us; Nexus 5 Build/JOP40D) AppleWebKit/535.19 (KHTML, like Gecko) Chrome/18.0.1025.166 Mobile Safari/535.19\"}}}" -DfitnesseSuiteToRun=HsacExamples.SlimTests.BrowserTests.ScriptTest
```
ps the map fixture you suggested doesn't work, chromeOptions is really picky on what you have supply

### CODE REVIEW

File: src/main/java/nl/hsac/fitnesse/junit/selenium/SeleniumJsonGridDriverFactoryFactory.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    protected Map<String, Object> getCapabilities() {
+        String capabilitiesString = getProperty(SELENIUM_JSONCAPABILITIES);
+        try {
+            Map<String, Object> result = new HashMap<String, Object>();
+            Gson gson = new Gson();
+            result = (Map<String, Object>)gson.fromJson(capabilitiesString, result.getClass());
+            return result;
+        } catch (Exception e) {
+            throw new RuntimeException("Unable to parse Selenium capabilities: " + capabilitiesString
+                    + "\n*", e);
```
Comment (fhoeben) [inline src/main/java/nl/hsac/fitnesse/junit/selenium/SeleniumJsonGridDriverFactoryFactory.java]: Why the `+ "\n*"`?

**Comment by fhoeben** [comment]
What is the problem you experienced with `JsonHelper`? Is that something that must/can be addressed?

Maybe if Gson is the answer that should be used by JsonHelper, and then the factory could use that?

File: src/main/java/nl/hsac/fitnesse/junit/selenium/SeleniumJsonGridDriverFactoryFactory.java (hunk 1 above)
Comment (teunisnl) [inline src/main/java/nl/hsac/fitnesse/junit/selenium/SeleniumJsonGridDriverFactoryFactory.java]: Left over of the Copy & Paste

**Comment by teunisnl** [comment]
Issue with Jsonhelper that does not support Maps within Maps (where object is another Map<string, object>)
if I use ```|connect to driver at|${GRID_HUB}          |with json capabilities |{"browserName":"chrome", "chromeOptions":{"mobileEmulation":{"deviceName":"Google Nexus 5"}}}|``` I get an error, while this is good json param.

so fixing/changing the jsonhelper might be better option. maby there better option then gson (I couldn't find a better, less dependent solution)

**Comment by teunisnl** [comment]
Not that I need it, but for acceptance testing I could add a import/export json to map fixture

**Comment by fhoeben** [comment]
Your change to JsonHelper broke a unit test, I believe the new setup no longer formats one json property per line.

Furthermore it would be good to add unit test to verify the Json parser can handle nested Maps.

**Comment by teunisnl** [comment]
Added unit test,
reverted back to old json for formating gson need an expected class to format

**Comment by fhoeben** [comment]
I managed to make your test pass without using gson. Is that also an acceptable solution, or are the other reasons to introduce it?
See https://github.com/fhoeben/hsac-fitnesse-fixtures/tree/mobileEmulationOnGrid

**Comment by teunisnl** [comment]
ill run this monday against the grid to check if chrome is still happy

**Comment by teunisnl** [comment]
Oke tested it on chrome, as expected no issues.

Also tested your map fixture fix, that works too.

**Pull request merged**